app/src/main/java/com/codepath/travel/fragments/TripListFragment.java

```diff
@@ -84,22 +92,22 @@ public void onViewCreated(View view, Bundle savedInstanceState) {
 
     public void populateTrips() {
         if (mUserId == null) {
+            showNoTripsFound();
             return;
         }
         mTrips.clear();
         Trip.getAllTripsForUser(mUserId, fetchUser, (trips, e) -> {
-            if (e == null) {
-                mTrips.addAll(trips);
-                mTripsAdapter.notifyDataSetChanged();
-            } else {
-                Log.d(TAG, String.format("Failed to populate all trips: %s", e.getMessage()));
-            }
+            resetTripAdapter(trips, e);
         });
     }
 
-    public void setUser(String userId) {
-        mUserId = userId;
-        populateTrips();
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if (!(context instanceof TripClickListener)) {
+            throw new RuntimeException("Activity should implement ImagePickerFragmentListener");
```

Comment:
Nice. (Don't forget to update "ImagePickerFragmentListener" to "TripClickListener")